velox/vector/BaseVector.h

```diff
@@ -412,9 +412,7 @@ class BaseVector {
       const BaseVector* source,
       vector_size_t targetIndex,
       vector_size_t sourceIndex,
-      vector_size_t count) {
-    VELOX_NYI();
-  }
+      vector_size_t count) = 0;
```

Comment:
This works. Another way could be to use VELOX_UNSUPPORTED("Only flat vectors support copy operation.").

I'm seeing we do something like this already, but throwing std::runtime_error instead of  VELOX_UNSUPPORTED. It would be nice to unify.

```
virtual const BufferPtr& values() const {
    throw std::runtime_error("Only flat vectors have a values buffer");
  }
``